compiler/src/dotty/tools/repl/Rendering.scala

```diff
@@ -52,6 +49,7 @@ private[repl] class Rendering(parentClassLoader: Option[ClassLoader] = None):
       }
 
       myClassLoader = new AbstractFileClassLoader(ctx.settings.outputDir.value, parent)
+      val maxPrintElements = ctx.settings.VreplMaxPrintElements.valueIn(ctx.settingsState)
```

Comment:
If we plan to have 2 separate flags for the limits of elements and characters then this setting would actually correspond to the limit of characters, not the limit of elements, so we should give the setting the proper name from the start instead of renaming it later

Reply:
As mentioned above my intention was to keep this PR small, but yes you raise a valid point, so I'll add the split into two properties (with different name and semantics) into this PR. Some time later today - I'll let you know.

Reply:
Ok sure - done via 9f5cc905a935d09f04fb31cca03336a2b9b3a25a
Note that your comment rather applied to `Rendering.truncate` below, where I just renamed `maxPrintElements` to `maxPrintCharacters`.

Reply:
I meant doing the rename of every occurance of `Element` to `Character` and `element` to `character` in the scope of this PR - not just renaming this single variable

Reply:
Just to double check: you're aware that the second argument to `ScalaRunTime.stringOf` is `maxElements` and not `maxCharacters`?
https://scala-lang.org/api/3.x/scala/runtime/ScalaRunTime$.html
![grafik](https://user-images.githubusercontent.com/506752/191734749-f4e636b9-a74b-48db-9604-80b945feb765.png)

I mean, if you want to, sure I'll rename it. I just think it wouldn't really make sense. It'd essentially be something like
```scala
val maxPrintCharacters = ctx.settings.VreplMaxPrintElements.valueIn(ctx.settingsState)
ScalaRunTime.stringOf(value, maxElements = maxPrintCharacters)
```
Either way, please confirm.

Reply:
> did I get the entire idea wrong?

No certainly not, your explanation above looks completely correct, up until (but excluding) this sentence: 
> An element of a string is a character so `stringOf(value, maxElements = maxPrintCharacters)` would still make sense.

An element of a string certainly is a character. But the `value` in this case is _not_ a string. It's a `List[Int]`, and `stringOf` inspects that list and converts the them into a String, up until the `maxElements` index. 
```scala
scala> scala.runtime.ScalaRunTime.stringOf(List(10,20,30,40), maxElements = 10)
val res0: String = List(10, 20, 30, 40)

scala> scala.runtime.ScalaRunTime.stringOf(List(10,20,30,40), maxElements = 4)
val res1: String = List(10, 20, 30, 40)

scala> scala.runtime.ScalaRunTime.stringOf(List(10,20,30,40), maxElements = 3)
val res2: String = List(10, 20, 30)

scala> scala.runtime.ScalaRunTime.stringOf(List(10,20,30,40), maxElements = 2)
val res3: String = List(10, 20)
```

Does that make sense?
Otherwise, it's really no big deal, I can rename that variable, and we clean it up in the followup PR that splits up the one setting into two (`-Vrepl-max-print-characters` and `-Vrepl-max-print-elements`).

Reply:
Ok, I think the point is that somehow I missed this comment, sorry for that

> Correct, the value is used in two places with different semantics, namely as the cut-off value for 'number of elements to be rendered' as well as 'number of characters to be displayed'. That doesn't really make sense, but it was like that before, and I wanted to keep this PR as simple as possible and only make the previously hard-coded value configurable.

I would say that in this situations the reasonable options would be to either
1) add both `-Vrepl-max-print-characters` and `-Vrepl-max-print-elements` settings in the same PR

or

2) add just one setting in this PR and replace the hardcoded value only in the places where it semantically corresponds to this setting and after that add the other setting in a followup PR

I still think we should not introduce a single setting changing these two values (maxElements and maxCharacters) just to later change the semantics of this setting to set just one value instead of two

Reply:
In the interest in keeping this PR small, let's go with Option 2. 
I reverted to the hardcoded value where it has different semantics (i.e. where it truncates by string length). 

If you specify the setting when starting the repl it does work just fine (`scala -Vrepl-max-print-elements:20`), but dynamically updating the setting value in the repl (e.g. via `:settings -Vrepl-max-print-elements:20`) does not have an affect. Hence, the ScriptedTest fails. 

I tried a few things - e.g. using reflection: given that the invocation of `stringOf` happens via reflection, I tried to do the same with the setting, but for some reason it doesn't find even the method:

```scala
val settingClazz = Class.forName("dotty.tools.dotc.config.Settings$Setting", true, myClassLoader)
val valueInMethod = settingClazz.getMethod("valueIn", classOf[SettingsState])
valueInMethod.invoke(ctx.settings.VreplMaxPrintElements, ctx.settingsState)

// java.lang.NoSuchMethodException: 
// dotty.tools.dotc.config.Settings$Setting.valueIn(dotty.tools.dotc.config.Settings$SettingsState)
```

Any idea what's going on, or alternative ways to fix it? I'm running out of ideas tbh.

Reply:
It looks like the problem is that when `classLoader` is evaluated, the `else` branch gets executed only once at the beginning so `myReplStringOf` is not set again with a different value of maxElements. The solution might be to change `myReplStringOf` to a function taking 2 arguments: the value to print and the limit